### PR TITLE
Deprecation of some API

### PR DESCRIPTION
**Pull request by julienlancelot**

### CODE REVIEW

File: sonar-plugin-api/src/main/java/org/sonar/api/Property.java [hunk 1]
```diff
@@ -109,7 +109,9 @@
    * by its key.
    *
    * @since 3.3
+   * @deprecated since 6.1, as it was not used and too complex to maintain.
    */
+  @Deprecated
   String propertySetKey() default "";
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/Property.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

File: sonar-plugin-api/src/main/java/org/sonar/api/PropertyField.java [hunk 2]
```diff
@@ -55,7 +55,10 @@
   /**
    * Indicative size of the field value in characters. This size is not validated, it is merely used by the GUI
    * to size the different input fields of a property set.
+   *
+   * @deprecated since 6.1, as it was only used for UI.
    */
+  @Deprecated
   int indicativeSize() default 20;
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/PropertyField.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

File: sonar-plugin-api/src/main/java/org/sonar/api/config/PropertyDefinition.java [hunk 3]
```diff
@@ -299,6 +299,10 @@ public boolean multiValues() {
     return multiValues;
   }
 
+  /**
+   * @deprecated since 6.1, as it was not used and too complex to maintain.
+   */
+  @Deprecated
   public String propertySetKey() {
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/config/PropertyDefinition.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

File: sonar-plugin-api/src/main/java/org/sonar/api/config/PropertyDefinition.java [hunk 4]
```diff
@@ -503,6 +507,10 @@ public Builder multiValues(boolean multiValues) {
       return this;
     }
 
+    /**
+     * @deprecated since 6.1, as it was not used and too complex to maintain.
+     */
+    @Deprecated
     public Builder propertySetKey(String propertySetKey) {
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/config/PropertyDefinition.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

File: sonar-plugin-api/src/main/java/org/sonar/api/config/PropertyFieldDefinition.java [hunk 5]
```diff
@@ -124,6 +126,10 @@ public Builder description(String description) {
       return this;
     }
 
+    /**
+     * @deprecated since 6.1, as it was only used for UI.
+     */
+    @Deprecated
     public Builder indicativeSize(int indicativeSize) {
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/config/PropertyFieldDefinition.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

File: sonar-plugin-api/src/main/java/org/sonar/api/config/PropertyFieldDefinition.java [hunk 6]
```diff
@@ -81,6 +79,10 @@ public String description() {
     return description;
   }
 
+  /**
+   * @deprecated since 6.1, as it was only used for UI.
+   */
+  @Deprecated
   public int indicativeSize() {
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/config/PropertyFieldDefinition.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

**Comment by SonarTech** [comment]
SonarQube analysis reported 10 issues
* MINOR 4 minor
* INFO 6 info

Watch the comments in this conversation to review them.

#### 4 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MINOR [PropertyDefinition.java#L145](https://github.com/SonarSource/sonarqube/blob/1756c146fdeea958f1463231c73616c725fa9501/sonar-plugin-api/src/main/java/org/sonar/api/config/PropertyDefinition.java#L145): Remove this use of "propertySetKey"; it is deprecated. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. MINOR [PropertyDefinition.java#L145](https://github.com/SonarSource/sonarqube/blob/1756c146fdeea958f1463231c73616c725fa9501/sonar-plugin-api/src/main/java/org/sonar/api/config/PropertyDefinition.java#L145): Remove this use of "propertySetKey"; it is deprecated. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. MINOR [PropertyFieldDefinition.java#L57](https://github.com/SonarSource/sonarqube/blob/1756c146fdeea958f1463231c73616c725fa9501/sonar-plugin-api/src/main/java/org/sonar/api/config/PropertyFieldDefinition.java#L57): Remove this use of "indicativeSize"; it is deprecated. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. MINOR [PropertyFieldDefinition.java#L57](https://github.com/SonarSource/sonarqube/blob/1756c146fdeea958f1463231c73616c725fa9501/sonar-plugin-api/src/main/java/org/sonar/api/config/PropertyFieldDefinition.java#L57): Remove this use of "indicativeSize"; it is deprecated. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

File: server/sonar-server/src/main/java/org/sonar/server/component/ws/EventsWs.java [hunk 7]
```diff
@@ -36,10 +36,11 @@ public void define(Context context) {
   }
 
   private void defineIndexAction(NewController controller) {
-    controller.createAction("index")
+    WebService.NewAction action = controller.createAction("index")
       .setDescription("Documentation of this web service is available <a href=\"http://redirect.sonarsource.com/doc/old-web-service-api.html\">here</a>")
       .setResponseExample(getClass().getResource("events-index-example.json"))
       .setSince("2.6")
       .setHandler(RailsHandler.INSTANCE);
+    RailsHandler.addFormatParam(action);
```
Comment (teryk) [inline server/sonar-server/src/main/java/org/sonar/server/component/ws/EventsWs.java]: what does it do?

File: server/sonar-server/src/main/java/org/sonar/server/user/ws/FavouritesWs.java [hunk 8]
```diff
@@ -22,11 +22,13 @@
...
 
-public class FavoritesWs implements WebService {
+import static org.sonar.api.server.ws.RailsHandler.addFormatParam;
+
+public class FavouritesWs implements WebService {
 
   @Override
   public void define(Context context) {
-    NewController controller = context.createController("api/favorites");
+    NewController controller = context.createController("api/favourites");
```
Comment (teryk) [inline server/sonar-server/src/main/java/org/sonar/server/user/ws/FavouritesWs.java]: checks `routes.rb`. I think that we want to handle backward compatibility and the 2 should work

File: sonar-plugin-api/src/main/java/org/sonar/api/Property.java [hunk 9]
```diff
@@ -109,7 +109,9 @@
    * by its key.
    *
    * @since 3.3
+   * @deprecated since 6.1, as it was not used and too complex to maintain.
```
Comment (teryk) [inline sonar-plugin-api/src/main/java/org/sonar/api/Property.java]: 👍

File: server/sonar-server/src/main/java/org/sonar/server/component/ws/EventsWs.java (hunk 7 above)
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/component/ws/EventsWs.java]: It's adding format parameter (it was missing)

File: server/sonar-server/src/main/java/org/sonar/server/user/ws/FavouritesWs.java (hunk 8 above)
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/user/ws/FavouritesWs.java]: routes.rb is using 'favourite' : https://github.com/SonarSource/sonarqube/blob/master/server/sonar-web/src/main/webapp/WEB-INF/config/routes.rb#L6.

I've badly use 'favorites' when creating the documentation, but no one complains before :(

**Pull request merged**